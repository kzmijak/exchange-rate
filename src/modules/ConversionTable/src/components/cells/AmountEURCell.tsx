import { FC } from "react";
import { ConversionTableRow } from "../../models/ConversionTableRow";
import { Typography } from "@mui/joy";
import { useCurrencyConverter } from "modules/CurrencyConversion";
import { columnsConfig } from "../../config/columnsConfig";

type AmountEURCellProps = { row: ConversionTableRow };

export const AmountEURCell: FC<AmountEURCellProps> = ({ row }) => {
  const { convertPlnToEur } = useCurrencyConverter();

  return (
    <td>
      <Typography textAlign={columnsConfig.AmountEUR.contentAlign}>
        {convertPlnToEur(row.amountPLN)}
      </Typography>
    </td>
  );
};
