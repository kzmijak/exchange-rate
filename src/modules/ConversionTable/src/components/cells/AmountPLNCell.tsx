import { FC } from "react";
import { Box, Typography } from "@mui/joy";
import { ConversionTableRow } from "../../models/ConversionTableRow";

type AmountPLNCellProps = { row: ConversionTableRow };

export const AmountPLNCell: FC<AmountPLNCellProps> = ({ row }) => {
  return (
    <Box component="td">
      <Typography textAlign="right">{row.amountPLN}</Typography>
    </Box>
  );
};
