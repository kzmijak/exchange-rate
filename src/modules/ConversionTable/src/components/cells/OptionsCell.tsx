import { Box, Button } from "@mui/joy";
import { FC } from "react";
import { ConversionTableRow } from "../../models/ConversionTableRow";

type OptionsCellProps = {
  row: ConversionTableRow;
};

export const OptionsCell: FC<OptionsCellProps> = ({ row }) => {
  const handleClick = () =>
    alert(`I REALLY WANT TO DELETE THIS WHOLE GUY ${row.title} `);
  return (
    <Box component="td">
      <Button onClick={handleClick}>Delete</Button>
    </Box>
  );
};
