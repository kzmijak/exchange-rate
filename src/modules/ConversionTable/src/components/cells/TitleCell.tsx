import { FC } from "react";
import { Typography } from "@mui/joy";
import { ConversionTableRow } from "../../models/ConversionTableRow";

type TitleCellProps = { row: ConversionTableRow };

export const TitleCell: FC<TitleCellProps> = ({ row }) => {
  return (
    <td>
      <Typography>{row.title}</Typography>
    </td>
  );
};
