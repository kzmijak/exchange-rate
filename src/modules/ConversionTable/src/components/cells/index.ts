export { TitleCell } from "./TitleCell";
export { AmountEURCell } from "./AmountEURCell";
export { AmountPLNCell } from "./AmountPLNCell";
export { OptionsCell } from "./OptionsCell";
