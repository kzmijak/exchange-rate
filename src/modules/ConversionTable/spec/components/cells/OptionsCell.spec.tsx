import { describe, it, expect, vi } from "vitest";
import { screen } from "@testing-library/react";
import { OptionsCell } from "../../../src/components/cells/OptionsCell";
import { row1 } from "../../__fixtures__/rows";
import { renderWithApi } from "../../__fixtures__/renderWithApi";
import userEvent from "@testing-library/user-event";

describe("TitleCell", () => {
  const Component = () => <OptionsCell row={row1} />;
  const deleteButtonDOM = () => screen.getByRole("button", { name: "Delete" });

  it("should match snapshot", () => {
    expect(renderWithApi(<Component />).container).toMatchSnapshot();
  });

  it("should contain Delete button", async () => {
    const deleteRow = vi.fn();
    renderWithApi(<Component />, { deleteRow });
    expect(deleteButtonDOM()).toBeInTheDocument();
    await userEvent.click(deleteButtonDOM());
    expect(deleteRow).toHaveBeenCalledOnce();
  });
});
