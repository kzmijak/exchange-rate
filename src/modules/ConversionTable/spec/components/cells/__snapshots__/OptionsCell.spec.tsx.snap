// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TitleCell > should match snapshot 1`] = `
<div>
  <td
    align="center"
  >
    <button
      class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeSm css-1gh95q6-JoyButton-root"
      type="button"
    >
      Delete
    </button>
  </td>
</div>
`;
